eclipse-collections/src/main/java/org/eclipse/collections/impl/set/mutable/SetAdapter.java

```diff
@@ -144,6 +144,16 @@ public boolean containsAll(Collection<?> collection)
     @Override
     public boolean equals(Object obj)
     {
+        if (obj == this)
+        {
+            return true;
+        }
+
+        if (!(obj instanceof Set))
+        {
+            return false;
+        }
```

Comment:
Same here. The delegate is a Set, so it's contractually obligated to perform the same check.